app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js

```diff
@@ -79,6 +80,16 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       }
     },
 
+    getCanonicalHref: function () {
+      var link = document.querySelector('link[rel=canonical]')
+
+      if (link) {
+        return link.href
+      }
+
+      return undefined
```

Comment:
I think a JS function will return `undefined` by default, so this line might not be needed? Might be worth having a test to check this behaviour as well.

Reply:
Thanks - I've replaced it with `this.nullValue` as that's what other functions are using, but can remove that line entirely if you think I should, as `this.nullValue` still just resolves to `undefined`.

Reply:
I think remove it if it's safe to remove it - no need having it if it's not doing something.

Reply:
Will do - thanks! 👍